modules/internal/ChapelArray.chpl

```diff
@@ -107,90 +107,55 @@ module ChapelArray {
   // with a privatized value that can be retrieved by the pid
   // without communication.
   proc _newPrivatizedClass(value) : int {
-
-    const n = numPrivateObjects.fetchAdd(1);
-
+    const pid = numPrivateObjects.fetchAdd(1);
     const hereID = here.id;
     const privatizeData = value.dsiGetPrivatizeData();
-    on Locales[0] do
-      _newPrivatizedClassHelp(value, value, n, hereID);
+    const definedConst = if isSubtype(value.type, BaseDom) then value.definedConst
+                                                           else false;
 
-    proc _newPrivatizedClassHelp(parentValue, originalValue, n, hereID) {
-      var newValue = originalValue;
+    coforall loc in Locales do on loc {
+      var newValue;
       if hereID != here.id {
-        newValue = parentValue.dsiPrivatize(privatizeData);
-        if isSubtype(parentValue.type, BaseDom) {
-          newValue.definedConst = parentValue.definedConst;
+        newValue = value.dsiPrivatize(privatizeData);
+        if isSubtype(value.type, BaseDom) {
+          newValue.definedConst = definedConst;
         }
-        __primitive("chpl_newPrivatizedClass", newValue, n);
-        newValue.pid = n;
       } else {
-        __primitive("chpl_newPrivatizedClass", newValue, n);
-        newValue.pid = n;
-      }
-      cobegin {
-        if chpl_localeTree.left._instance != nil then
-          on chpl_localeTree.left do
-            _newPrivatizedClassHelp(newValue, originalValue, n, hereID);
-        if chpl_localeTree.right._instance != nil then
-          on chpl_localeTree.right do
-            _newPrivatizedClassHelp(newValue, originalValue, n, hereID);
+        newValue = value;
       }
+      newValue.pid = pid;
+      __primitive("chpl_newPrivatizedClass", newValue, pid);
     }
-
-    return n;
+    return pid;
   }
 
   // original is the value this method shouldn't free, because it's the
   // canonical version. The rest are copies on other locales.
-  proc _freePrivatizedClass(pid:int, original:object):void
-  {
+  proc _freePrivatizedClass(pid:int, original:object) : void  {
     // Do nothing for null pids.
     if pid == nullPid then return;
 
-    on Locales[0] {
-      _freePrivatizedClassHelp(pid, original);
-    }
-
-    proc _freePrivatizedClassHelp(pid, original) {
+    coforall loc in Locales do on loc {
       var prv = chpl_getPrivatizedCopy(unmanaged object, pid);
       if prv != original then
         delete prv;
 
       extern proc chpl_clearPrivatizedClass(pid:int);
       chpl_clearPrivatizedClass(pid);
 
-      cobegin {
-        if chpl_localeTree.left._instance != nil then
-          on chpl_localeTree.left do
-            _freePrivatizedClassHelp(pid, original);
-        if chpl_localeTree.right._instance != nil then
-          on chpl_localeTree.right do
-            _freePrivatizedClassHelp(pid, original);
-      }
     }
   }
 
-  proc _reprivatize(value) {
+  proc _reprivatize(value) : void {
```

Comment:
Curious, did you add this just for completeness? Should we remove all of them instead?

Reply:
Yeah, this was the only privatization routine that didn't have the return type, though https://github.com/Cray/chapel-private/issues/3663 makes me consider removing the void return type specifications.

Reply:
For internal routines, it doesn't matter whether we use `: void` or not.  For user-facing documentation, my preference would be to be sure to declare return types for all procedures that have one and have lack of a return type mean `void` (and create tools/scripts to keep us honest on that point).  I can imagine I may lose that argument.  I don't object to `: void` here in either case given that it's not user-facing (though like Engin, if I'd been reviewing, I probably would've asked).